### ISSUE TITLE
[Documentation] Inconsistency between AddCache() and AddCaches()

### ISSUE THREAD

**Issue by abatishchev** [AzureAD/microsoft-identity-web#1412]
### Documentation related to component

Extensions methods for Token cache 

### Please check all that apply

- [x] typo
- [ ] documentation doesn't exist
- [x] documentation needs clarification
- [ ] error(s) in the example
- [ ] needs an example

### Description of the issue

Here's an article: https://docs.microsoft.com/en-gb/azure/active-directory/develop/msal-net-token-cache-serialization?tabs=aspnet#in-memory-token-cache
Also the reference: https://docs.microsoft.com/en-us/dotnet/api/microsoft.identity.web.tokencacheproviders.inmemory.inmemorytokencacheproviderextension.addinmemorytokencaches?view=azure-dotnet-preview

They all mention `AddXXXCaches()` as the extension method names (plural).

However TokenCacheExtensions.cs [the latest main](*https://github.com/AzureAD/microsoft-identity-web/blob/master/src/Microsoft.Identity.Web/TokenCacheProviders/TokenCacheExtensions.cs) as well as the published to NuGet version `1.16.0` uses `AddXXXCache()`  (singular).

Please update the docs.

**Comment by abatishchev** [comment]
I see. So one is this:

```c#
public static IConfidentialClientApplication AddInMemoryTokenCache (this IConfidentialClientApplication confidentialClientApp);
```

means is for `IConfidentialClientApplication confidentialClientApp`.

And the another is this:

```c#
public static IServiceCollection AddInMemoryTokenCaches (this IServiceCollection services);
```

means is for `IServiceCollection services`.

It's very confusing but now makes sense.

**Comment by jmprieur** [comment]
but you're right that the article is wrong? https://docs.microsoft.com/en-gb/azure/active-directory/develop/msal-net-token-cache-serialization?tabs=aspnet#in-memory-token-cache

**Comment by abatishchev** [comment]
Hi Jean-Marc,

From the perspective of `ApplicatiinBuilder`, the article is correct. I didn't use the extensions methods it describes though, can't speak about the substance. But on the surface it looks okay.

One minor issue: the section called `Available serialization technologies` must rather be called `Available caching technologies`or something like that. Nothing to do with serialization per se, right?

However, from the perspective of `IServiceCollection`, the article is poor as it has no mentions of the respective extension methods whatsoever. The corresponding `/// xmldoc` is more helpful than the doc.

To summarize:
- The doc covers app builder
- Has small typo
- Doesn't cover service collection (ASP.NET et al)

**Comment by jmprieur** [comment]
@abatishchev 
did you see the tabs in the document ?

![image](https://user-images.githubusercontent.com/13203188/131204750-4adcd659-9c35-41ca-bc02-1180caf3e66e.png)

**Comment by abatishchev** [comment]
Indeed I didn't. Now it makes even more sense. Doesn't work for me still, but I'll figure this out. Thanks!

**Comment by jmprieur** [comment]
@abatishchev : I proposed a pr to fix the doc article:
https://github.com/MicrosoftDocs/azure-docs-pr/pull/170701
